File: GoLang/radixsort/radixsort.go

package radixsort

func RadixSort(slice []int) {

  if len(slice) == 0 {
    return
  }

  length := len(slice)
  higher := slice[0]
  exp := 1
  var temp []int

  for index := 0; index < length; index++ {
    temp = append(temp, 0)
    if slice[index] > higher {
      higher = slice[index]
    }
  }

  for higher/exp > 0 {
    var bucket [10]int

    for index := 0; index < length; index++ {
      bucket[(slice[index]/exp) % 10] = bucket[(slice[index]/exp) % 10] + 1
    }

    for index := 1; index < 10; index++ {
      bucket[index] = bucket[index] + bucket[index-1]
    }

    for index := length-1; index >= 0; index-- {
      bucket[(slice[index] / exp) % 10] = bucket[(slice[index] / exp) % 10] - 1
      temp[bucket[(slice[index] / exp) % 10]] = slice[index]
    }

    for index := 0; index < length; index++ {
      slice[index] = temp[index]
    }

    exp = exp * 10
  }
}

File: src/javascript/RadixSort.js
/**
 * @param {number[]} vector 
 */
function RadixSort(vector) {
    let i = 0;
    let b = new Int32Array(vector.length);
    let bigger = vector[0];
    let exp = 1;

    for (i = 0; i < vector.length; i++) {
        if (vector[i] > bigger)
            bigger = vector[i];
    }

    while(parseInt(bigger / exp) > 0){
        let bucket = new Int32Array(10);

        for (i = 0; i < vector.length; i++)
            bucket[parseInt(parseInt((vector[i] / exp)) % 10)]++;
        for (i = 1; i < 10; i++)
            bucket[i] += bucket[i - 1];
        for (i = vector.length - 1; i >= 0; i--)
            b[--bucket[parseInt(parseInt((vector[i] / exp)) % 10)]] = vector[i];
        for (i = 0; i < vector.length; i++)
            vector[i] = b[i];
        
        exp *= 10;
    }
}

function main() {
    const MAX_LENGTH = 20;
    let vector = [];
    let values = "[";
    for (let index = 0; index < MAX_LENGTH; index++) {
        let value = Math.floor(Math.random() * 99);  // 0 - 99
        vector.push(value);
        values += `${value}, `;
    }
    values += "]";

    console.log(values);

    RadixSort(vector);
    values = "[";
    vector.forEach(v => values+=`${v}, `);
    values += "]";
    console.log(values);
}

main();